### PR TITLE
Move bank tests to separate file

### PR DESCRIPTION
**Pull request by bw-solana**
#### Problem
See #29400 . `bank.rs` is/was a monstrosity and unwieldy

#### Summary of Changes
Moved tests mod out of `bank.rs` and into new file `bank/tests.rs`
There was also a `runtime/tests/bank.rs` file that had a single unit test. I moved this test into `bank/tests.rs` and deleted `runtime/tests/bank.rs`

### CODE REVIEW

File: runtime/src/bank.rs [hunk 1]
```diff
@@ -193,6 +193,8 @@ mod address_lookup_table;
 mod builtin_programs;
 mod metrics;
 mod sysvar_cache;
+#[cfg(test)]
+pub(crate) mod tests;
```
Comment (t-nelson) [inline runtime/src/bank.rs:197]:
```suggestion
pub(self) mod tests;
```

`self` _should_ be sufficient

**Review by t-nelson — COMMENTED** [review]
thanks for taking the plunge here! much needed

File: runtime/src/bank.rs (hunk 1 above)
Comment (bw-solana) [reply runtime/src/bank.rs:197]: there's 1 case where `bank_forks` reaches in here to call `update_vote_account_timestamp`

Comment (t-nelson) [reply runtime/src/bank.rs:197]: oh god...  maybe pop it out of the `tests` module, then?

**Review by t-nelson — DISMISSED** [review]
:shipit:

**Comment by t-nelson** [comment]
latest push was just a rebase?

**Comment by bw-solana** [comment]
> latest push was just a rebase?

Yes, but there were conflicts that had to be resolved:
https://github.com/solana-labs/solana/commit/d048a1903f1875282d8a79c70dc7e7cfee93b2c1#diff-ed47b4a0198313377e091bb3957bbbc63d937805426d1b2b6de39d0a50d32a0cR12907-R12912

https://github.com/solana-labs/solana/commit/090b990e15abb06a51befac398b97173c0328a52#diff-ed47b4a0198313377e091bb3957bbbc63d937805426d1b2b6de39d0a50d32a0c

**Review by t-nelson — COMMENTED** [review]
> > latest push was just a rebase?
> 
> Yes, but there were conflicts that had to be resolved (minor Bank test updates + 1 new bank test added):
> https://github.com/solana-labs/solana/commit/d048a1903f1875282d8a79c70dc7e7cfee93b2c1#diff-ed47b4a0198313377e091bb3957bbbc63d937805426d1b2b6de39d0a50d32a0cR12907-R12912
> 
> https://github.com/solana-labs/solana/commit/090b990e15abb06a51befac398b97173c0328a52#diff-ed47b4a0198313377e091bb3957bbbc63d937805426d1b2b6de39d0a50d32a0c

ok, thanks!  r+ approval from @taozhu-chicago that the new test was correctly handled

**Review by taozhu-chicago — APPROVED** [review]
Thanks for lifting this -- the changes are so huge, Github had trouble to load diff for me 😨 The new test I added for removing congestion multiplier looks good in its new home. 

A minor suggestion, why stop here? why not break the `runtime/src/bank/tests.rs` into several smaller pieces? perhaps grouping by testing functions/areas?

**Comment by bw-solana** [comment]
> A minor suggestion, why stop here? why not break the `runtime/src/bank/tests.rs` into several smaller pieces? perhaps group by testing functions/areas?

I like it. I can pursue this as a follow-up

**Comment by mvines** [comment]
A backport to 1.15 seems fine. I don't think we should backport to v1.14 though, so you'll still have some rebase pain there Tao unfortunately

**Pull request merged**